src/serac/physics/integrators/displacement_hyperelastic_integrator.cpp

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "serac/physics/integrators/displacement_hyperelastic_integrator.hpp"
+
+#include "serac/infrastructure/profiling.hpp"
+#include "serac/numerics/expr_template_ops.hpp"
+#include "serac/numerics/array_4D.hpp"
+#include "serac/physics/utilities/physics_utils.hpp"
+
+namespace serac::mfem_ext {
+
+void DisplacementHyperelasticIntegrator::CalcKinematics(const mfem::FiniteElement&    element,
+                                                        const mfem::IntegrationPoint& int_point,
+                                                        mfem::ElementTransformation& parent_to_reference_transformation)
+{
+  // Calculate the reference to stress-free transformation
+  CalcInverse(parent_to_reference_transformation.Jacobian(), dxi_dX_);
+
+  // Calculate the derivatives of the shape functions in the reference space
+  element.CalcDShape(int_point, dN_dxi_);
+
+  // Calculate the derivatives of the shape functions in the stress free configuration
+  Mult(dN_dxi_, dxi_dX_, dN_dX_);
+
+  // Calculate the displacement gradient using the current DOFs
+  MultAtB(input_state_matrix_, dN_dX_, du_dX_);
+
+  solid_util::calcDeformationGradient(du_dX_, F_);
+
+  // Calculate the inverse of the deformation gradient
+  mfem::CalcInverse(F_, Finv_);
+
+  // Calculate the B matrix (dN/Dx where x is the current configuration)
+
+  if (geom_nonlin_) {
+    // If we're including geometric nonlinearities, we integrate on the current deformed configuration
+    mfem::Mult(dN_dX_, Finv_, B_);
+    det_J_ = F_.Det();
+  } else {
+    // If not, we integrate on the undeformed stress-free configuration
+    B_     = dN_dX_;
+    det_J_ = 1.0;
+  }
+}
+
+double DisplacementHyperelasticIntegrator::GetElementEnergy(
```

Comment:
Again, probably better to say "GetHyperelasticElementEnergy", then you could mix material models in the same mesh, by using a mask

Reply:
This name is unfortunately fixed as it plugs into the MFEM inheritance structure. This is partially why I tagged this integrator as specific to hyperelasticity.